split_to_CMYK.jsx

```diff
@@ -1 +1,124 @@
+
+		var layerY = layers.add();
+  			layerY.name = 'Yellow';
+  			layerY.blendMode = BlendMode.MULTIPLY;
+
+		var layerM = layers.add();
+  			layerM.name = 'Magenta';
+  			layerM.blendMode = BlendMode.MULTIPLY;
+
+		var layerC = layers.add();
+  			layerC.name = 'Cyan';
+  			layerC.blendMode = BlendMode.MULTIPLY;
+}
+  //used in splitChannels function
+function selectCopy() {
+        doc.selection.selectAll();
+        doc.selection.copy();
+  }
+//copies channels into corresponding layers
+function splitChannels(targetChannel) {
+    doc.activeLayer = doc.artLayers.getByName("Background");
+    doc.activeChannels = [doc.channels.getByName(targetChannel)];
+    selectCopy();
+    doc.activeLayer = doc.artLayers.getByName(targetChannel);
+    doc.paste();
+}
+
+function addLevelsHueSat(layerName, hue, sat, light) {
+//add levels adj layer
+    doc.activeLayer = doc.artLayers.getByName(layerName);
+    var c2t = function (s) {
+        return app.charIDToTypeID(s);
+    };
+    var s2t = function (s) {
+        return app.stringIDToTypeID(s);
+    };
+    var descriptor = new ActionDescriptor();
+    var descriptor2 = new ActionDescriptor();
+    var descriptor3 = new ActionDescriptor();
+    var reference = new ActionReference();
+
+    reference.putClass( s2t( "adjustmentLayer" ));
+    descriptor.putReference( c2t( "null" ), reference );
+    descriptor2.putString( s2t( "name" ), (layerName) + """ Levels""" );
+    descriptor2.putBoolean( s2t( "group" ), true );
+    descriptor3.putEnumerated( s2t( "presetKind" ), s2t( "presetKindType" ), s2t( "presetKindDefault" ));
+    descriptor2.putObject( s2t( "type" ), s2t( "levels" ), descriptor3 );
+    descriptor.putObject( s2t( "using" ), s2t( "adjustmentLayer" ), descriptor2 );
+    executeAction( s2t( "make" ), descriptor, DialogModes.NO );
+//add hue sat adj layer - if statement is so Black doesnt get a hue adj layer
+    if (layerName != 'Black') {
+    var c2t = function (s) {
+        return app.charIDToTypeID(s);
+    };
+    var s2t = function (s) {
+        return app.stringIDToTypeID(s);
+    };
+    var descriptor = new ActionDescriptor();
+    var descriptor2 = new ActionDescriptor();
+    var descriptor3 = new ActionDescriptor();
+    var reference = new ActionReference();
+
+    reference.putClass( s2t( "adjustmentLayer" ));
+    descriptor3.putReference( c2t( "null" ), reference );
+    descriptor.putString( s2t( "name" ), (layerName) + """ color adjust""" );
+    descriptor.putBoolean( s2t( "group" ), true );
+    descriptor2.putEnumerated( s2t( "presetKind" ), s2t( "presetKindType" ), s2t( "presetKindDefault" ));
+    descriptor2.putBoolean( s2t( "colorize" ), true );
+    descriptor.putObject( s2t( "type" ), s2t( "hueSaturation" ), descriptor2 );
+    descriptor3.putObject( s2t( "using" ), s2t( "adjustmentLayer" ), descriptor );
+    executeAction( s2t( "make" ), descriptor3, DialogModes.NO );
+//sets hue, sat, lightness
+    var descriptor = new ActionDescriptor();
```

Comment:
Maybe try to rework some of these `var` names so they're easier to follow?  I'm having a hard time digesting these auto-generated functions (which always makes me nervous when it comes to future maintainability of your code).